### PR TITLE
BigQuery: add missing explict coverage for '_helpers'

### PR DESCRIPTION
**Pull request by tseaver**

### CODE REVIEW

File: bigquery/google/cloud/bigquery/_helpers.py [hunk 1]
```diff
@@ -388,7 +344,18 @@ def _should_retry(exc):
     return reason == 'backendError' or reason == 'rateLimitExceeded'
 
 
-def get_sub_prop(container, keys, default=None):
+DEFAULT_RETRY = retry.Retry(predicate=_should_retry)
```
Comment (tswast) [inline bigquery/google/cloud/bigquery/_helpers.py]: I wonder if we should move this to a public module? Currently these docs won't get rendered.

File: bigquery/google/cloud/bigquery/_helpers.py [hunk 2]
```diff
@@ -388,7 +344,18 @@ def _should_retry(exc):
...
+
+Any method with a ``retry`` parameter will be retried automatically,
+with reasonable defaults. To disable retry, pass ``retry=None``.
+To modify the default retry behavior, call a ``with_XXX`` method
+on ``DEFAULT_RETRY``. For example, to change the deadline to 30 seconds,
+pass ``retry=bigquery.DEFAULT_RETRY.with_deadline(30)``.
+"""
+
+
+def _get_sub_prop(container, keys, default=None):
```
Comment (tswast) [inline bigquery/google/cloud/bigquery/_helpers.py]: Do we need a leading underscore for functions in a private module?

Comment (tseaver) [reply bigquery/google/cloud/bigquery/_helpers.py]: FWIW, every other function in the module follows that convention.

Comment (tswast) [reply bigquery/google/cloud/bigquery/_helpers.py]: Fine with me either way.

File: bigquery/google/cloud/bigquery/_helpers.py (hunk 1 above)
Comment (tseaver) [reply bigquery/google/cloud/bigquery/_helpers.py]: See PR #5552.

**Pull request merged**